quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/services/resources/QuarkusWelcomeResource.java

```diff
@@ -16,209 +16,24 @@
-            }
-        } catch (IOException e) {
-            throw new WebApplicationException(e, Response.Status.INTERNAL_SERVER_ERROR);
-        }
-    }
-
-    private Response createWelcomePage(String successMessage, String errorMessage) {
-        try {
-            Theme theme = getTheme();
-
-            if(Objects.isNull(theme)) {
-                Log.error("Theme is null please check the \"--spi-theme-default\" parameter");
-                errorMessage = "The theme is null";
-                ResponseBuilder rb = Response.status(Status.BAD_REQUEST)
-                        .entity(errorMessage)
-                        .cacheControl(CacheControlUtil.noCache());
-                return rb.build();
-            }
-
-            Map<String, Object> map = new HashMap<>();
-
-            map.put("adminConsoleEnabled", isAdminConsoleEnabled());
-            map.put("productName", Version.NAME);
-
-            map.put("properties", theme.getProperties());
-            map.put("adminUrl", session.getContext().getUri(UrlType.ADMIN).getBaseUriBuilder().path("/admin/").build());
-
-            map.put("resourcesPath", "resources/" + Version.RESOURCES_VERSION + "/" + theme.getType().toString().toLowerCase() +"/" + theme.getName());
-            map.put("resourcesCommonPath", "resources/" + Version.RESOURCES_VERSION + "/common/keycloak");
+public class QuarkusWelcomeResource extends WelcomeResource {
 
-            boolean bootstrap = shouldBootstrap();
-            map.put("bootstrap", bootstrap);
-            if (bootstrap) {
-                boolean isLocal = isLocal();
-                map.put("localUser", isLocal);
-
-                String localAdminUrl = getLocalAdminUrl();
-
-                map.put("localAdminUrl", localAdminUrl);
-                map.put("adminUserCreationMessage", "or set the environment variables KEYCLOAK_ADMIN and KEYCLOAK_ADMIN_PASSWORD before starting the server");
-
-                if (isLocal) {
-                    String stateChecker = setCsrfCookie();
-                    map.put("stateChecker", stateChecker);
-                }
-            }
-            if (successMessage != null) {
-                map.put("successMessage", successMessage);
-            }
-            if (errorMessage != null) {
-                map.put("errorMessage", errorMessage);
-            }
-            FreeMarkerProvider freeMarkerUtil = session.getProvider(FreeMarkerProvider.class);
-            String result = freeMarkerUtil.processTemplate(map, "index.ftl", theme);
-
-            ResponseBuilder rb = Response.status(errorMessage == null ? Status.OK : Status.BAD_REQUEST)
-                    .entity(result)
-                    .cacheControl(CacheControlUtil.noCache());
-            return rb.build();
-        } catch (Exception e) {
-            throw new WebApplicationException(e, Response.Status.INTERNAL_SERVER_ERROR);
-        }
+    @Override
+    protected String getAdminCreationMessage() {
```

Comment:
I think we can change the message on `WelcomeResource` and remove more code from this class. The message from `WelcomeResource` does not make sense anymore.